legal-api/src/legal_api/services/filings/validations/agm_extension.py

```diff
@@ -39,6 +41,15 @@ def validate(business: Business, filing: Dict) -> Optional[Error]:
     if business.legal_type not in enabled_filings:
         return Error(HTTPStatus.FORBIDDEN,
                      [{'error': babel(f'{business.legal_type} does not support agm extension filing.')}])
+    authorized_permissions = PermissionService.get_authorized_permissions_for_user()
+    if validate_certify_name(filing):
+        allowed_role_comments = ListActionsPermissionsAllowed.EDITABLE_CERTIFY_NAME.value
+        if allowed_role_comments not in authorized_permissions:
+            return Error(
+                HTTPStatus.FORBIDDEN,
+                [{ 'message': f'Permission Denied - You do not have permissions to change certified by in this filing.'}]
+            )
+    
```

Comment:
You could move the entire check into a function since its been reused in multiple places. Something like

```suggestion
    if not is_certify_by_editable(filing):
            return Error(
                HTTPStatus.FORBIDDEN,
                [{ 'message': f'Permission Denied - You do not have permissions to change certified by in this filing.'}]
            )
    
```